pkg/cloud/aws/actuators/machine_scope.go

```diff
@@ -131,3 +134,46 @@ func (m *MachineScope) Close() {
 		klog.Errorf("[machinescope] failed to store provider status for machine %q in namespace %q: %v", m.Machine.Name, m.Machine.Namespace, err)
 	}
 }
+
+func machineConfigFromProviderSpec(clusterClient client.MachineClassesGetter, providerConfig clusterv1.ProviderSpec) (*v1alpha1.AWSMachineProviderSpec, error) {
+	var config v1alpha1.AWSMachineProviderSpec
+	if providerConfig.Value != nil {
+		klog.V(4).Info("Decoding ProviderConfig from Value")
+		return unmarshalProviderSpec(providerConfig.Value)
+	}
+
+	if providerConfig.ValueFrom != nil && providerConfig.ValueFrom.MachineClass != nil {
+		ref := providerConfig.ValueFrom.MachineClass
+		klog.V(4).Info("Decoding ProviderConfig from MachineClass")
+		if ref != nil {
```

Comment:
we already know `providerConfig.ValueFrom.MachineClass` is != nil, this check can go away since it's on line 145